pkg/serviceaccount/legacy.go

```diff
@@ -126,6 +140,32 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			klog.V(4).Infof("Service account UID no longer matches %s/%s: %q != %q", namespace, serviceAccountName, string(serviceAccount.UID), serviceAccountUID)
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
+			warning.AddWarning(ctx, "", "Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.")
+			today := time.Now().Format("2006-01-02")
+			if secret.GetAnnotations()[LastUsedAnnotationKey] != today || secret.GetLabels()[InUseLabelKey] != today {
```

Comment:
```suggestion
			if secret.Annotations[LastUsedAnnotationKey] != today || secret.Labels[InUseLabelKey] != today {
```

Reply:
do we need to tolerate the annotation/label having a value up to one day in the future, so that API servers with a few seconds time skew don't fight each other at midnight?

Reply:
we don't need to. to tolerate value up to one day, the value should be a timestamp instead of a date. when the timestamp is recorded, there will be time skew as well.

Reply:
Also, should we explicitly make the time.Now() be UTC, or is that the default?

Reply:
done. 

> Leaving the annotation/label alone if it is one day ahead of time.Now() would let us avoid that.

the side effect is that we can possibly skip one update. e.g. two requests arriving to the same API server on 2022-08-02 and 2022-08-03. the second request will not be updated.